eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/log/EventLog.scala

```diff
@@ -575,7 +575,7 @@ abstract class EventLog[A <: EventLogState](id: String) extends Actor with Event
         // 100% filtering coverage for certain replication network topologies.
         acc
       case (acc, e) =>
-        snr += 1L
+        snr = (snr + 1) max e.vectorTimestamp.value.getOrElse(id, 0)
```

Comment:
I actually tried something different. I moved the sequence number adjustment down to [line 585](https://github.com/RBMHTechnology/eventuate/blob/wip-286-disaster-recovery-enhancements/eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/log/EventLog.scala#L585):

        (updated, clock.copy(sequenceNr = snr max lvv.value.getOrElse(id, 0), versionVector = lvv))

Not sure if it is better than your proposal, but as far as I understand it should not suffer from the problem you described and it is certainly much simpler. @kongo2002 In any case this all results in potential gaps in all event log implementations including the leveldb one, but currently we have a _gap-detecion_ algorithm in there that makes all event iterations with gaps fail. It was introduced with #274. I wonder if we should simply drop this gap detection or have to find something smarter. WDYT?